action/src/index.ts

```diff
@@ -547,6 +551,11 @@ export const main = async ({
         .map((s) => s.trim())
         .filter((s) => s !== '');
       return globList;
+    } else if (env.TARGET_DIR) {
+      log.log(
+        `##[info] Removing all files from target dir ${env.TARGET_DIR} on target branch`
+      );
+      return [`${env.TARGET_DIR}/*`, '!.git'];
```

Comment:
we probably want to make sure that we recursively delete everything from the target dir (and not just the immediate files)